RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+  }
+});
+```
+
+**Amplify v5 DX**
+
+https://user-images.githubusercontent.com/70438514/222475304-128de96a-9a06-4fcc-8f0d-8f19592ae9eb.mov
+
+<br />
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+type customChannel = "custom_channel";
+
+// Each key in the map represents a payload event and the key value is the data type for that event.
+// Note: If an event is assigned the null type, the payload object will not contain a data key.
+type customEventDataMap = {
+  A: number;
+  B: string;
+  C: null;
+  D: Object
+};
+
+Hub.dispatch<customChannel, customEventDataMap >("custom_channel", {
+  event: "A",
+  data: 42
+});
+
+Hub.listen<customChannel, customEventDataMap>("custom_channel", ({payload}) => {
+  switch(payload.event) {
+    case "A":
+      payload.data;
+      break;
+    case "B":
+      payload.data;
+      break;
+    case "C":
+      payload.data;
+      break;
+    case "D":
+      payload.data;
+      break;
+  }
+});
+```
+
+**Proposed Amplify v6 DX**
+
+https://user-images.githubusercontent.com/70438514/222475311-36b4f8bc-d338-4ad1-9910-5d3518fb71c0.mov
+
+## TypeScript support for Amplify Configuration
+
+To help customers configure Amplify categories, we are introducing type support for the `Amplify.configure` API. The examples below demonstrate an `Auth` configuration.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+const authConfig = {
+  userPoolId: "us-east-1_0yqxxHm5q",
+  userPoolClientId: "3keodiqtm52nhh2ls0vQfs5v1q",
+  signUpVerificationMethod: "code"
+};
+
+Amplify.configure({
+  Auth: authConfig
+});
+```
+**Amplify v5 DX**
+
+<img width="494" alt="configure-v5" src="https://user-images.githubusercontent.com/70438514/222543613-50124bf6-a5d8-4b6f-9196-320100922ec7.png">
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+const authConfig : AuthConfig = {
+  userPoolId: "us-east-1_0yqxxHm5q",
+  userPoolClientId: "3keodiqtm52nhh2ls0vQfs5v1q",
+  signUpVerificationMethod: "code"
+};
+
+Amplify.configure({
+  Auth: authConfig
+});
+```
+**Proposed Amplify v6 DX**
+
+https://user-images.githubusercontent.com/70438514/222543618-d5c42798-28c5-41e9-9ff6-c9061b86ef43.mov
+
+<br />
+
+Try out the new types here: https://www.typescriptlang.org/play#gist/7332e86effa3b0d627eebfc71fad1aca
+
+# `Auth` Category Changes
+
+Amplify is proposing the following changes for the `Auth` category. Similar changes will be applied across all of the `Auth` APIs but examples for specific APIs are highlighted below.
+
+## TypeScript support for user attributes
+
+User attributes inference on the `signUp` API.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+
+Auth.signUp({
+  username: "username",
+  password: "*******",
+  attributes: {
+    email: "email@domain.com"
+  }
+});
+
+```
+
+**Amplify v5 DX**
+
+![signup-v5](https://user-images.githubusercontent.com/70438514/222456624-b4af7349-db9d-4304-9cdc-08cf649e2d31.png)
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+
+signUp({
```

Comment:
We should add proposed import path for all v6 modular change snippets. Those feedbacks would very helpful.

Reply:
Yeah we struggled with this on the RFC, we wanted to keep this RFC more about types instead of functional APIs & mono-repo API changes (which is why API & Storage still use `API.` & `Storage.`), need to make more consistent though